org.opentosca.container.core/src/org/opentosca/container/core/engine/impl/ToscaEngineServiceImpl.java

```diff
@@ -371,7 +371,11 @@ public boolean isOperationOfRelationshipBoundToSourceNode(final CSARID csarID, f
     @Override
     public String getRelatedNodeTemplateID(final CSARID csarID, final QName serviceTemplateID,
                                            final String nodeTemplateID, final QName relationshipType) {
-
+        return this.getRelatedNodeTemplateIDs(csarID, serviceTemplateID, nodeTemplateID, relationshipType).stream().findFirst().orElse(null);        
```

Comment:
instead of streaming again, why not write:

```
List<String> relatedIds = getRelatedNodeTemplateIDs(...);
return relatedIds.isEmpty() ? null : relatedIds.get(0);
```

Reply:
I like the tenary more than functional stuff myself but if we had all history here (i mean also the SVN history from way back..) you would see that we we're not allowed to do it (although I'm pretty sure there are still some tenaries somewhere ;) )

Leave as is